pandas/_testing/asserters.py

```diff
@@ -399,9 +399,8 @@ def _get_ilevel_values(index, level):
     # skip exact index checking when `check_categorical` is False
     if check_exact and check_categorical:
         if not left.equals(right):
-            diff = (
-                np.sum((left._values != right._values).astype(int)) * 100.0 / len(left)
-            )
+            thesum = np.sum((left._values != right._values).astype(int))
+            diff = thesum * 100.0 / len(left)
```

Comment:
why is this changed?

Reply:
Because when you do `astype(int)`, you 

> is it worth posting the results of a test file with reveal_types for this?

Can you point me to an example?

Reply:
Must have had to change it in a previous commit with an earlier version of numpy.  Reverted in next commit.